File: admin/src/main/java/top/zzf4/blog/entity/model/Article.java
package top.zzf4.blog.entity.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Article {

    private Long id;

    private String title;

    private String content;

    private Integer categoryId;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime publishDate;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updateDate;

    private Integer views;

    private Integer wordNum;

    private List<Tag> tags;

    private String thumbnailUrl; // 缩略图url
}

File: src/types/Article.ts
export interface Article<T> {
  id?: number;
  title: string;
  content: string;
  categoryId: number;
  categoryName?: string;
  createTime?: string;
  updateTime?: string;
  tags?: T[];
  views?: number;
  wordNum?: number; // TypeScript中没有byte类型，通常使用number代替
  thumbnailUrl?: string;
}